docs/data/material/experimental-api/css-theme-variables/usage.md

```diff
@@ -0,0 +1,169 @@
+# Usage
+
+<p class="description">Learn how to use the new experimental APIs to adopt CSS theme variables.</p>
+
+## Getting started
+
+`Experimental_CssVarsProvider` is a new provider that generates CSS theme variables and attach a reference to the theme object (a React context).
+
+```js
+import { Experimental_CssVarsProvider as CssVarsProvider } from '@mui/material/styles';
+
+function App() {
+  return <CssVarsProvider>...</CssVarsProvider>;
+}
+```
+
+Once the `App` renders on the screen, you will see the CSS theme variables in the html `:root` stylesheet. The variables are flatten and prefixed with `--mui` by default:
+
+```css
+/* generated global stylesheet */
+:root {
+  --mui-palette-primary-main: #1976d2;
+  --mui-palette-primary-light: #42a5f5;
+  --mui-palette-primary-dark: #1565c0;
+  --mui-palette-primary-contrastText: #fff;
+  /* ...other variables */
+}
+```
+
+:::info
+💡 The `CssVarsProvider` is built on top of the [`ThemeProvider`](/material-ui/customization/theming/#themeprovider) with extra features like CSS variables generation, storage synchronization, unlimited color schemes, etc.
+:::
+
+## Toggle between light and dark mode
+
+The `useColorScheme` hook lets you read and update the user selected mode:
+
+```jsx
+import {
+  Experimental_CssVarsProvider as CssVarsProvider,
+  useColorScheme,
+} from '@mui/material/styles';
+
+// ModeSwitcher is an example interface that users use to toggle between modes.
+// Material UI does not provide the toggle interface, you have to build it yourself.
+const ModeSwitcher = () => {
+  const { mode, setMode } = useColorScheme();
+  const [mounted, setMounted] = React.useState(false);
+
+  React.useEffect(() => {
+    setMounted(true);
+  }, []);
+
+  if (!mounted) {
+    // for server-side rendering
+    // learn more at https://github.com/pacocoursey/next-themes#avoid-hydration-mismatch
+    return null;
+  }
+
+  return (
+    <Button
+      variant="outlined"
+      onClick={() => {
+        if (mode === 'light') {
+          setMode('dark');
+        } else {
+          setMode('light');
+        }
+      }}
+    >
+      {mode === 'light' ? 'Dark' : 'Light'}
+    </Button>
+  );
+};
+
+function App() {
+  return (
+    <CssVarsProvider>
+      <ModeSwitcher />
+    </CssVarsProvider>
+  );
+}
+```
+
+## Using the theme variables
+
+- `theme.vars` (recommended): an object that refers to the CSS theme variables.
+
+  ```js
+  const Button = styled('button')(({ theme }) => ({
+    backgroundColor: theme.vars.palette.primary.main, // var(--mui-palette-primary-main)
+    color: theme.vars.palette.primary.contrastText, // var(--mui-palette-primary-contrastText)
+  }));
+  ```
+
+  For **TypeScript**, the typings are not enabled by default. You can follow the [typescript setup](#typescript) to enable the typings.
+
+  :::warning
+  Make sure that the components accessing `theme.vars.*` are rendered under the new provider, otherwise you will get `TypeError`.
```

Comment:
```suggestion
  Make sure that the components accessing `theme.vars.*` are rendered under the new provider, otherwise you will get a `TypeError`.
```